core-plugins/src/test/java/co/cask/hydrator/plugin/batch/source/FileBatchSourceTest.java

```diff
@@ -463,4 +478,149 @@ public void testRecursiveTimeFilterRegex() throws Exception {
     Assert.assertTrue(outputValue.contains("Hello,World"));
     Assert.assertTrue(outputValue.contains("CDAP,Platform"));
   }
-}
+
+  @Test
+  public void testFileBatchInputFormatText() throws Exception {
+    File fileText = new File(temporaryFolder.newFolder(), "test.txt");
+    String outputDatasetName = "test-filesource-text";
+
+    Schema textSchema = Schema.recordOf("file.record",
+                                        Schema.Field.of("offset", Schema.of(Schema.Type.LONG)),
+                                        Schema.Field.of("body", Schema.nullableOf(Schema.of(Schema.Type.STRING))),
+                                        Schema.Field.of("file", Schema.nullableOf(Schema.of(
+                                          Schema.Type.STRING))));
+    ApplicationManager appManager = createSourceAndDeployApp(fileText, "text", outputDatasetName, textSchema);
+
+    FileUtils.writeStringToFile(fileText, "Hello,World!");
+
+    workflowStartAndWait(appManager);
+
+    List<StructuredRecord> expected = ImmutableList.of(
+      StructuredRecord.builder(textSchema)
+        .set("offset", (long)0)
+        .set("body", "Hello,World!")
+        .set("file", fileText.toURI().toString())
+        .build()
+    );
+
+    DataSetManager<Table> outputManager = getDataset(outputDatasetName);
+    List<StructuredRecord> output = MockSink.readOutput(outputManager);
+
+    Assert.assertEquals(output, expected);
+  }
+
+  @Test
+  public void testFileBatchInputFormatAvro() throws Exception {
+    File fileAvro = new File(temporaryFolder.newFolder(), "test.avro");
+    String outputDatasetName = "test-filesource-avro";
+
+    ApplicationManager appManager = createSourceAndDeployApp(fileAvro, "avro", outputDatasetName, RECORD_SCHEMA);
```

Comment:
you should have avro and parquet tests that specify a schema that is different, but compatible to, write schema. This is to test the logic that is setting the input schema key in the input format. For example, I believe you should be able to remove a field from the write schema and it should read it correctly.